### ISSUE TITLE
Allow read-only mode

### ISSUE THREAD

**Issue by Yyuzu** [GroundApps/ShoppingList#59]
It could be useful to have a readonly mode.
You implement this with 2 passwords, one allowing more than the other ;-)

**Comment by ghost** [comment]
2 passwords is not a good idea, because of the api.
Better, a little icon, like a lock on touch you can edit

**Comment by Yyuzu** [comment]
I meant more for being able to share with someone who can only read but not alter the list ^^

But after a bit more thinking, maybe it's not what I wanted. I'd like to avoid misuse and clean of the list by mistake ^^
I don't know what could prevent this.

**Comment by the0ne** [comment]
you could send the list by using the [share] button and then choose your email-program.
the receiver will not be able to misuse or clean your list and won't even need to install the client.

**Comment by Yyuzu** [comment]
This is not really what I had in mind, I was going for an external access whenever the user wants to, without requiring me to manually share the list.

**Comment by the0ne** [comment]
@Kyoshiro-san, would that (read-only allowed) user access the list via the ShoppingList App or just via the browser? If only via the browser then I propose to create an additional readonly.php that can be accessed without having to enter a password and that can only be used to view the list. I further propose this new readonly.php to be disabled by default and it would have to be enabled explicitly via a config file entry.
Please comment if that is what you need or further clarify your request.

**Comment by Yyuzu** [comment]
Yeah using the browser could be enough with a share link even (random string?), no need for a password in fact ^^
Sorry for the delay in the answer.

**Comment by the0ne** [comment]
I had totally missed the notification about your message - sorry for the delay!
Would you want that random string to change over time (e.g. every few days), or would it be a fixed random string (e.g. a one-way hash of the password).

**Comment by the0ne** [comment]
can be tested here: https://github.com/GroundApps/ShoppingList_Backend/pull/82
